github-runner-provisioner/macm1.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+type runnerConfig struct {
+	imageId              string
+	hostResourceGroupArn string
+	placement            types.Placement
+	instanceCount        int32
+	shutdownBehavior     types.ShutdownBehavior
+	instanceType         types.InstanceType
+}
+
+var macM1HostResourceGroupArn = "arn:aws:resource-groups:us-east-1:914373874199:group/GitHub-Runners"
+
+var macM1Config = runnerConfig{
+	imageId: "ami-01b8fcd5770ceb9c1",
+	placement: types.Placement{
+		HostResourceGroupArn: &macM1HostResourceGroupArn,
+	},
+	instanceCount:    1,
+	shutdownBehavior: "terminate",
+	instanceType:     "mac2.metal",
+}
+
+func macRunnerUserData(ctx context.Context, owner string, repo string) (string, error) {
+	token, err := getGitHubRunnerToken(ctx, owner, repo)
+	if err != nil {
+		return "", err
+	}
+
+	const labels = "macOS-arm64"
```

Comment:
I wonder if we should put the OS version as a label also? I could see a world where we want to test on 12.5, 12.4 etc

Reply:
That makes sense, but I'd like to do that on a new PR.
I think we could have options like:
`macOS-arm64-12.5`
`macOS-arm64-latest`